### PR TITLE
Improve it tests

### PR DESCRIPTION
**Pull request by jacek-lewandowski**
Several changes have been made in this patch. To name a few:
- Bumped Scala 2.11 compiler to the newest version 2.11.7
- Reenabled Spark REPL tests
- Integration test classes always get the configuration from SparkTemplate.defaultConf
- SparkTemplate defaultConf is now a method which returns a clone of the config
- SparkTemplate alows to disable SparkContext and doesn't start it implicitly
- Fixed SparkTemplate problem that SparkContext was unnecesarily restarted sometimes
- Cassandra runner looks for a free range of ports
- Cassandra runner stops uncessary instances immediately
- Tuned some memory requirements
- Integration tests generates their schema in parallel in order to minimise table creation fixed delays
- Data is generated using executeAsync rather than execute
- Ignored CheckpointStreamSpec as it showed flaky, especially in parallel runs (to be addressed in a separate ticket)
- Each test uses a dedicated keyspace, whose name is consistently generated from test class name
- Disabled durable writes for testing
- Tests were groupped so that at least repl, streaming, auth, ssl are run in separate processes
- Tests run in parallel, the parallelisation level is automatically resolved according to the amount of OS memory (roughly we need 3 GB per single test task)

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -214,28 +246,71 @@ object Settings extends Build {
     publish in (IntegrationTest,packageBin) := ()
   )
 
+  def makeTestGroups(tests: Seq[TestDefinition]) = {
```
Comment (pkolaczk) [inline project/Settings.scala]: for a complex method like this an explicit return type would improve readability

File: spark-cassandra-connector-embedded/scala-2.11/src/main/scala/com/datastax/spark/connector/embedded/SparkRepl.scala [hunk 2]
```diff
@@ -28,12 +27,21 @@ trait SparkRepl {
     }
 
     val interp = new SparkILoop(Some(in), new PrintWriter(out))
-    org.apache.spark.repl.Main.interp = interp
+    Main.interp = interp
+    // due to some mysterious compilation error classServer needs to be accessed via reflection
+    //      Main.classServer.start()
```
Comment (pkolaczk) [inline spark-cassandra-connector-embedded/scala-2.11/src/main/scala/com/datastax/spark/connector/embedded/SparkRepl.scala]: would be nice to get more info about that mysterious compilation error

File: spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/DynamicCassandraPorts.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  private lazy val basePath = Paths.get(sys.props("baseDir"), "target", "ports")
+
+  private lazy val basePorts = 10000 until 20000 by 50
+
+  private def portLockFile(port: Int) = basePath.resolve(s"port_$port.lock")
+
+  def apply(): DynamicCassandraPorts = DynamicCassandraPorts(chooseBasePort())
+
+  def verifyRange(host: String, ports: Seq[Int]): Boolean = {
```
Comment (pkolaczk) [inline spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/DynamicCassandraPorts.scala]: Minor nitpick: The method name does not make it obvious what the result means. It has no side effect, so using a verb is also not needed. `areAllPortsAvailable`?

File: spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/UserDefinedProperty.scala [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package com.datastax.spark.connector.embedded
+
+import java.net.InetAddress
+
+import org.apache.commons.configuration.ConfigurationException
+
+object UserDefinedProperty {
```
Comment (pkolaczk) [inline spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/UserDefinedProperty.scala]: De ja vu. Was it moved from somewhere?

File: spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/SparkCassandraITFlatSpecBase.scala [hunk 5]
```diff
@@ -13,4 +20,41 @@ trait SparkCassandraITWordSpecBase extends WordSpec with SparkCassandraITSpecBas
...
+    val className = this.getClass.getSimpleName
+    val suffix =  StringUtils.splitByCharacterTypeCamelCase(className.filter(_.isLetterOrDigit)).mkString("_")
+    s"test_$suffix".toLowerCase()
+  }
+
+  val ks = getKsName
+
+  implicit val ec = SparkCassandraITSpecBase.ec
+
+  def doAsync(units: Future[Unit]*): Unit = {
```
Comment (pkolaczk) [inline spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/SparkCassandraITFlatSpecBase.scala]: A blocking method called "Async". I guess the purpose of this method is to just wait for all the futures to complete, right? So why not name it `awaitAll` ?

**Comment by pkolaczk** [comment]
Overall looks like a great improvement. +1

File: project/Settings.scala (hunk 1 above)
Comment (jacek-lewandowski) [inline project/Settings.scala]: ok

File: spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/DynamicCassandraPorts.scala (hunk 3 above)
Comment (jacek-lewandowski) [inline spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/DynamicCassandraPorts.scala]: ok

File: spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/UserDefinedProperty.scala (hunk 4 above)
Comment (jacek-lewandowski) [inline spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/UserDefinedProperty.scala]: I just extracted few classes from a single file into separate files.

File: spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/SparkCassandraITFlatSpecBase.scala (hunk 5 above)
Comment (jacek-lewandowski) [inline spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/SparkCassandraITFlatSpecBase.scala]: ok

**Comment by pkolaczk** [comment]
+1

**Pull request merged**